blockprotocol/blockprotocol
# Display stripe error message when upgrading an existing paid subscription

## 🌟 What is the purpose of this PR?




This PR handles stripe errors when upgrading an existing paid stripe subscription with an existing payment method.

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203623179643287/1203880489778056/f) _(internal)_


## 🛡 What tests cover this?



Manual testing

## ❓ How to test this?



1. Checkout the branch
2. Add the `NEXT_PUBLIC_BILLING_FEATURE_FLAG=1` env var to your `app/site/.env.local` to enable the billing feature flag
3. Run the local developer environment for the BP site
4. Checkout the HASH internal API repo on the `bw/upgrade-stripe-subscription-fixes` branch (if it hasn't merged yet)
5. Follow the instructions in the internal API README to run the local developer environment, except for the mongo db external service (as the BP site is already running mongo db)
6. Login as a user and navigate to `/settings/billing`
7. Ensure the logged in user is subscribed to the "hobby" subscription tier
8. Add the `4000000000000341` card as a new payment method (declines on second use)

<img width="858" alt="image" src="https://user-images.githubusercontent.com/42802102/222130638-fe1dfae4-b5eb-425e-bbed-8122066dc124.png">

9. Try upgrading the subscription, it should result in an error being displayed


## 📹 Demo



<img width="552" alt="image" src="https://user-images.githubusercontent.com/42802102/222129886-967a0112-ac67-45a9-95c0-06618d4a7eb0.png">
